code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -12,6 +13,12 @@ var StageProgress = React.createClass({
     largeDots: React.PropTypes.bool
   },
 
+  dotClicked(url) {
+    if (window.saveProgressAndNavigate) {
+      window.saveProgressAndNavigate(url);
```

Comment:
In general React tries to avoid calling globals and instead explicitly pass all data into the component (via `this.props`, [passed in here](https://github.com/code-dot-org/code-dot-org/blob/c41b3ccebbab9ac75a25906ad60ff63cc326e2b1/code-studio/src/js/progress.js#L156-L159)) or include it via a `require`.

Can `saveAnswers` be extracted to a helper that can be required from `levelGroup.js` and here?